File: Question 1-10/Q2/Q2.c
#include<stdio.h>
#include<stdlib.h>
void findrange(int * arr, int low, int high){
	int i, min = arr[low], max = arr[low], pos1 = low, pos2 = low;
	if(low==high){
		printf("%d\n",-1); // if the list is already sorted
		return;
	}
	for(i=low;i<=high;i++){
		if(min>arr[i]){
			min = arr[i];
			pos1 = i; //index of min
		}
		if(max<=arr[i]){
			max = arr[i];
			pos2 = i; //index of max
		}
	}
	if(pos1==low && pos2 == high)
		findrange(arr, low+1, high-1);
	else if(pos1==low)
		findrange(arr, low+1, high);
	else if(pos2==high)
		findrange(arr, low, high-1);
	else
		printf("%d %d\n",low, high);
}
int main(void){
	int * arr, n, i;
	// input
	scanf("%d",&n);
	arr = (int*)malloc(n*sizeof(int));
	for(i=0;i<n;i++)
		scanf("%d",&arr[i]);
	// recursive solution	
	findrange(arr, 0, n-1);
	return 0;
}

File: Question 1-10/Q2/Q2.java
import java.util.*;
public class Q2
{
    public static void findrange(int arr[], int low, int high){
        if(low==high){
            System.out.println(-1);
            return;
        }
        int min = arr[low], max = arr[low], pos1 = low, pos2 = low;
        for(int i=low;i<=high;i++){
            if(max<=arr[i]){
                max=arr[i];
                pos2=i;
            }
            if(min>arr[i]){
                min=arr[i];
                pos1=i;
            }
        }
        if(pos1==low && pos2==high)
            findrange(arr,low+1,high-1);
        else if(pos1==low)
            findrange(arr, low+1, high);
        else if(pos2==high)
            findrange(arr,low,high-1);
        else
            System.out.println(low+" "+high);
    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        Q2 obj = new Q2();
        int n;
        n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++)
            arr[i] = sc.nextInt();
            
        obj.findrange(arr,0,n-1);
    }
}
